doc/ref/states/requisites.rst

```diff
@@ -563,7 +563,7 @@ The ``onfail`` requisite is applied in the same way as ``require`` and ``watch``
     notify-build_failure:
       hipchat.send_message:
         - room_id: 123456
-        - message: "Building website fail on {{ salt.grains.get('id') }}"
```

Comment:
Disagree. Unless you are deliberately handling the `None` you _want_ it to raise an error rather than continue silently and then either fail somewhere else or do something bad. Therefore `[]` is preferred in the general case.

Reply:
Okay, get your point, better to have it throw an error if the index is missing in whatever['my_missing_key'] than have to deal with side-effects